crates/bevy_asset/src/info.rs

```diff
@@ -3,42 +3,56 @@ use bevy_utils::{HashMap, HashSet, Uuid};
 use serde::{Deserialize, Serialize};
 use std::path::PathBuf;
 
+/// Metadata for an asset source.
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct SourceMeta {
```

Comment:
How would you feel about removing this struct, in favor of just storing its field in `SourceInfo`?

Reply:
That would make sense. I take it that `SourceMeta`  was added for future proofing, but you could also add another field to `SourceInfo` if the need arises. I also wanted to rename `LoadContext::get_asset_metas` to `generate_asset_metas` and `Handle::as_weak` to `cast` or `recast` to make their intent clearer. cart said in #3522 that `as_weak` usage should probably be refactored in `bevy_text`, but it wasn't clear if the method should be removed or not.

Reply:
Could you do each of these changes? Use a separate commit for each, in case we need to revert.